source/server/http/admin.h

```diff
@@ -49,6 +49,9 @@ class AdminStreamImpl : public AdminStream {
   bool endStreamOnComplete() const override { return end_stream_on_complete_; }
 
 private:
+  // the callbacks member is a reference to a member of a ConnectionManagerImpl object. It is
+  // destroyed when connection is terminated. handlers using it should invalidate the reference by
+  // adding a method to be run when onDestroy() is invoked, using addOnDestroyCallback().
   const Http::StreamDecoderFilterCallbacks& callbacks_;
   const Http::HeaderMap& request_headers_;
```

Comment:
similar comments are also needed for request_headers_ and on_destroy_callbacks_.  Basically any time you have a reference or pointer as a member variable you need to comment on ownership & lifetime.